src/client/cl_main.cpp

```diff
@@ -1883,23 +1898,32 @@ static void CL_DownloadsComplete(void)
         CL_cURL_Shutdown();
         if (clc.cURLDisconnected)
         {
+#ifdef NEW_FILESYSTEM
+			clc.downloadRestart = qfalse;
+			clc.cURLReconnecting = qtrue;	// Don't reset attempted downloads
+#else
             if (clc.downloadRestart)
             {
                 if (!clc.activeCURLNotGameRelated) FS_Restart(clc.checksumFeed);
                 clc.downloadRestart = false;
             }
+#endif
             clc.cURLDisconnected = false;
             if (!clc.activeCURLNotGameRelated) CL_Reconnect_f();
             return;
         }
     }
 
+#ifndef NEW_FILESYSTEM
```

Comment:
I use a script to verify the ifdefs (no new filesystem changes outside them) so that's why I always use them. I expect you will probably remove the ifdefs altogether at some point if you end up using the filesystem.